courses/utils.py

```diff
@@ -0,0 +1,91 @@
+"""
+Utility functions for the courses app
+"""
+import re
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+
+NUMBER_SEASON_MAP = {
+    1: 'Spring',
+    2: 'Summer',
+    3: 'Fall'
+}
+
+
+def parse_edx_key(edx_course_key):
+    """
+    Attempts to parse a CourseRun's edx_course_key as an edX opaque key and return the portion
+    of the key that indicates the course's semester/year
+
+    Args:
+        edx_course_key (str): An edX course key (CourseRun.edx_course_key)
+
+    Returns:
+        str: The 'run' portion of a parsed CourseKey (opaque_keys.edx.keys.CourseKey#run),
+        eg: '1T2016'. Return None if the parse fails.
+    """
+    try:
+        course_key = CourseKey.from_string(edx_course_key)
+    except InvalidKeyError:
+        return None
+    return course_key.run if course_key else None
+
+
+def get_year_season_from_edx_key(parsed_course_key):
+    """
+    Verifies that a parsed course key matches our expected format and returns a year/season
+
+    Args:
+        parsed_course_key (str): A semester/year indicator from a course key (eg: '1T2016')
+
+    Returns:
+        tuple: (Year, Season number)
+    """
+    found = re.search('^[1-3]T[1|2][0-9]{3}$', parsed_course_key)
+    if found:
+        season_year_tuple = found.group().split('T')
+        return int(season_year_tuple[1]), int(season_year_tuple[0])
+    return ()
```

Comment:
Can you add an assertion to cover this case? I got to this path using `course-v1:MITX+MITx_Digital_learning_300+3Tabc`